contracts/otcMaker/OtcMaker.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+import { ECDSAUpgradeable } from "@openzeppelin/contracts-upgradeable/cryptography/ECDSAUpgradeable.sol";
+import { EIP712Upgradeable } from "@openzeppelin/contracts-upgradeable/drafts/EIP712Upgradeable.sol";
+import { FullMath } from "@uniswap/v3-core/contracts/libraries/FullMath.sol";
+
+import { PerpMath } from "@perp/curie-contract/contracts/lib/PerpMath.sol";
+import { PerpSafeCast } from "@perp/curie-contract/contracts/lib/PerpSafeCast.sol";
+import { IClearingHouse } from "@perp/curie-contract/contracts/interface/IClearingHouse.sol";
+import { IClearingHouseConfig } from "@perp/curie-contract/contracts/interface/IClearingHouseConfig.sol";
+import { IVault } from "@perp/curie-contract/contracts/interface/IVault.sol";
+import { IAccountBalance } from "@perp/curie-contract/contracts/interface/IAccountBalance.sol";
+
+import { SafeOwnable } from "../base/SafeOwnable.sol";
+
+import { IMerkleRedeem } from "../interface/IMerkleRedeem.sol";
+import { IOtcMaker } from "../interface/IOtcMaker.sol";
+import { ILimitOrderBook } from "../interface/ILimitOrderBook.sol";
+import { OtcMakerStorageV1 } from "../storage/OtcMakerStorage.sol";
+
+contract OtcMaker is SafeOwnable, EIP712Upgradeable, IOtcMaker, OtcMakerStorageV1 {
+    using PerpMath for int256;
+    using PerpMath for uint256;
+    using PerpSafeCast for int256;
+    using PerpSafeCast for uint256;
+
+    //
+    // MODIFIER
+    //
+
+    modifier onlyCaller() {
+        // OM_NC: not caller
+        require(_msgSender() == _caller, "OM_NC");
+        _;
+    }
+
+    //
+    // EXTERNAL NON-VIEW
+    //
+
+    function initialize(address clearingHouseArg, address limitOrderBookArg) external initializer {
+        __SafeOwnable_init();
+        _caller = _msgSender();
+        _clearingHouse = clearingHouseArg;
+        _limitOrderBook = limitOrderBookArg;
+        _vault = IClearingHouse(_clearingHouse).getVault();
+        _accountBalance = IClearingHouse(_clearingHouse).getAccountBalance();
+    }
+
+    function setCaller(address newCaller) external override onlyOwner {
+        // OM_ZA: zero address
+        require(newCaller != address(0), "OM_ZA");
+        _caller = newCaller;
+        emit UpdateCaller(_caller, newCaller);
+    }
+
+    function setMarginRatioLimit(uint24 marginRatioLimitArg) external override onlyOwner {
+        require(marginRatioLimitArg > 62500 && marginRatioLimitArg < 1000000, "OM_IMR");
```

Comment:
Is marginRatioLimitArg limited within 10%?
If yes, should modify `1000000` to `100000`.